osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -290,19 +296,54 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
                     currentChannelContainer.Clear(false);
                     currentChannelContainer.Add(loaded);
                     currentChannelContainer.FadeIn(500, Easing.OutQuint);
+
+                    if (messagePendingHighlight != null)
+                    {
+                        tryHighlightMessage(messagePendingHighlight);
+                        messagePendingHighlight = null;
+                    }
```

Comment:
Good idea, that simplified most of the highlighting logic. The only thing I'm unsure how to feel about is `Channel.HighlightedMessage` being set back to null once `DrawableChannel` receives the callback (to avoid highlighting again when, for example, reloading a channel by leaving and joining again), but I'll leave that to you as to whether you think it's fine and within what you have expected.

Reply:
I think I'd be more okay with the bindable flow if it was renamed to `LastHighlightedMessage` and was not set to `null`. Would that still work?

Reply:
I don't immediately see a reason why it wouldn't work personally.